app/components/atomic/galoy-input/galoy-redesigned-input.tsx

```diff
@@ -1,57 +1,110 @@
 import * as React from "react"
-import { TextInput } from "react-native"
+import { StyleProp, Text, TextInput, View, ViewStyle } from "react-native"
 
 import { Input, InputProps, makeStyles, useTheme } from "@rneui/themed"
 
 import { ComponentType } from "../../../types/jsx"
 
 const useStyles = makeStyles((theme) => ({
+  ContainerStyle: {
+    paddingBottom: 3,
+    paddingTop: 3,
+    marginLeft: 0,
+    borderRadius: 10,
+    backgroundColor: theme.colors.primary9,
+  },
   inputContainerFocused: {
     borderColor: theme.colors.primary5,
     backgroundColor: theme.colors.white,
-    shadowOffset: { width: 3, height: 2 },
-    shadowRadius: 2,
-    shadowOpacity: 1,
+    marginLeft: -7,
+    marginRight: -7,
   },
-  captionStyle: {
-    color: theme.colors.grey5,
-    textTransform: "lowercase",
+  errorStateStyle: {
+    borderColor: theme.colors.error5,
   },
 }))
 
 type GaloyInputProps = {
   initIsFocused?: boolean
   caption?: string
-}
+} & InputProps
 
 const GaloyInputFunctions = (
-  props: InputProps & GaloyInputProps,
+  props: GaloyInputProps,
   ref: React.Ref<TextInput> & React.Ref<React.PropsWithChildren<InputProps>>,
 ) => {
   const { theme } = useTheme()
-  const styles = useStyles(theme)
   const [isFocused, setIsFocused] = React.useState(props.initIsFocused ?? false)
+  const styles = useStyles(props)
+
+  return (
+    <View>
+      <LabelComponent
```

Comment:
What is the thinking behind using a custom LabelComponent instead of using the `labelStyle` and `labelProps` that are contained in the `Input` component? The same questions goes for the caption component. If it is necessary to use a CustomComponent, you can still pass that directly into the `label` prop of Input.